generic3g/MAPL_Generic.F90

```diff
@@ -239,6 +242,11 @@ subroutine gridcomp_get(gridcomp, unusable, &
       if (present(logger)) logger => outer_meta_%get_lgr()
       if (present(registry)) registry => outer_meta_%get_registry()
       if (present(geom)) geom = outer_meta_%get_geom()
+      if (present(num_levels)) then
```

Comment:
This needs discussion.   I remain concerned about "num_levels" being ambiguous, though less so in this context.

Possibly we just want the user to retrieve the vertical grid, like they would with geom.

Reply:
It can be easily be done but is there a need for users to retrieve anything other than the `num_levels`? We can rename it to `num_vertical_levels`.

Reply:
The ambiguity isn't whether it is "vertical" it is whether it is edges vs centers, grid vs field, ...

Nothing fundamental, but we want to get it "right" (TM) before we base too much code on it.

Reply:
Replaced the `num_level` accessor with `vertical_grid`.